### PR TITLE
Update istioctl install docs with overlay examples

### PR DESCRIPTION
**Pull request by ostromart**
Adds an advanced use section based on interactions with users in discuss.istio.io.

### CODE REVIEW

**Comment by ericvn** [comment]
Seems like the answer to https://github.com/istio/istio/issues/23206 might be to change the `istioctl manifest apply`'s to `instill install`'s.

I wish netlify was available to try and walk through the new items. Maybe just pull and build locally.

File: content/en/docs/setup/install/istioctl/index.md [hunk 1]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
 namespaces and enablement settings. The Istio community recommends using the `IstioOperator` API as it is more
 consistent, is validated, and follows the [community graduation process](https://github.com/istio/community/blob/master/FEATURE-LIFECYCLE-CHECKLIST.md#feature-lifecycle-checklist).
 
+## Advanced install customization
+
+### Customizing external charts and profiles
+
+`istioctl` `install`, `manifest generate` and `profile` commands can use any of the following sources for charts and
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:533]:
```suggestion
The `istioctl` `install`, `manifest generate` and `profile` commands can use any of the following sources for charts and
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 2]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+profiles:
+
+- compiled in charts. This is the default if no `--charts` option is set. The compiled in charts are the same as those
+in the `manifests/` directory in Istio release `.tgz`.
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:537]:
```suggestion
in the `manifests/` directory of the Istio release `.tgz`.
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 3]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+- charts in the local file system e.g. `istioctl install --charts istio-1.6.0/manifests`
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:538]:
```suggestion
- charts in the local file system, e.g., `istioctl install --charts istio-1.6.0/manifests`
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 4]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+- charts in GitHub e.g. `istioctl install --charts https://github.com/istio/istio/releases/download/1.6.0/istio-1.6.0-linux-arm64.tar.gz`
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:539]:
```suggestion
- charts in GitHub, e.g., `istioctl install --charts https://github.com/istio/istio/releases/download/1.6.0/istio-1.6.0-linux-arm64.tar.gz`
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 5]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+
+Local file system charts and profiles can be customized by editing the files in the `manifests/`. For extensive changes,
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:541]:
```suggestion
Local file system charts and profiles can be customized by editing the files in `manifests/`. For extensive changes,
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 6]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+it's recommended to make a copy of this directory and make changes to the copy. `istioctl` requires the same file layout
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:542]:
```suggestion
we recommend making a copy of the `manifests` directory and make changes there. Note, however, that the content layout
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 7]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+as in `manifests` to be preserved.
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:543]:
```suggestion
of the `manifests` file must be preserved.
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 8]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+can be referenced by name in the `IstioOperatorSpec` profile field. Built-in profiles are overlaid on the default profile YAML before user
+overlays are applied. For example, you can create a new profile file called `custom1.yaml` which customizes some settings
+from the `default` profile, and then apply a user overlay file on top of that:
+
+{{< text bash >}}
+$ istioctl generate --charts mycharts/ --set profile=custom1 -f path-to-user-overlay.yaml
+{{< /text >}}
+
+In this case, the `custom1.yaml` and `user-overlay.yaml` files will be overlaid on the `default.yaml` file to obtain the
+final values used as an input to manifest generation.
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:556]:
```suggestion
final values used as the input for manifest generation.
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 9]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+
+In general, creating new profiles is not necessary since a similar result can be achieved by passing multiple overlay
+files. For The command above is equivalent to passing two user overlay files:
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:559]:
```suggestion
files. For example, the command above is equivalent to passing two user overlay files:
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 10]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+
+{{< text bash >}}
+$ istioctl generate --charts mycharts/ --f manifests/profiles/custom1.yaml -f path-to-user-overlay.yaml
+{{< /text >}}
+
+Creating a custom profile is only required if you need to refer to a profile by name through the `IstioOperatorSpec`.
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:565]:
```suggestion
Creating a custom profile is only required if you need to refer to the profile by name through the `IstioOperatorSpec`.
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 11]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+
+### Patching the output manifest
+
+The `IstioOperator` CR input to `istioctl` is used to generate the output manifest containing the
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:569]:
```suggestion
The `IstioOperator` CR, input to `istioctl`, is used to generate the output manifest containing the
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 12]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+              # Deletion of map item
+              - path: spec.template.spec.containers.[name:discovery].securityContext
+          - kind: Service
+            name: istiod
+            patches:
+              - path: spec.ports.[name:https-dns].port
+                value: 11111 # OVERRIDDEN
+{{< /text >}}
+
+Passing the file to `istioctl manifest generate -f patch.yaml` applies the patches above to the default profile output
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:615]:
```suggestion
Passing the file to `istioctl manifest generate -f patch.yaml` applies the above patches to the default profile output
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 13]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v2
+              fieldPath: metadata.myPath
+        name: discovery
+        ports:
+        - containerPort: 1234
+
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:639]:
```suggestion
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 14]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+---
+
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:641]:
```suggestion
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 15]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:642]:
```suggestion
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 16]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+metadata:
+  name: istiod
+spec:
+  ports:
+  - name: https-dns
+    port: 11111
+---
+{{< /text >}}
+
+Note that the patches are applied in the given order, and each patch is applied over the resulting output from the
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:654]:
```suggestion
Note that the patches are applied in the given order. Each patch is applied over the output from the
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 17]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+previous patch. In addition, paths in patches that do not exist in the output manifest are created.
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:655]:
```suggestion
previous patch. Paths in patches that don't exist in the output manifest will be created.
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 18]
```diff
@@ -526,6 +526,139 @@ Some parameters will temporarily exist in both the Helm and `IstioOperator` APIs
...
+
+### List item path selection
+
+Both the `--set` flag and the `k8s.overlays` support list item selection by `[index]`, `[value]` or by `[key:value]`.
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md:659]:
```suggestion
Both the `istioctl --set` flag and the `k8s.overlays` field in `IstioOperator` CR support list item selection by `[index]`, `[value]` or by `[key:value]`.
```

File: content/en/docs/setup/install/istioctl/index.md (hunk 7 above)
Comment (ostromart) [reply content/en/docs/setup/install/istioctl/index.md:543]: it should be: of the `manifests` directory must be preserved.

Comment (frankbu) [reply content/en/docs/setup/install/istioctl/index.md:543]: Ah, sorry. See updated suggestion.

**Pull request merged**